src/services/anchor-service.ts

```diff
@@ -229,11 +247,11 @@ export default class AnchorService implements Contextual {
       let selected: Candidate = null;
 
       for (const pair of candidates) {
-        const record = (await this.ceramicService.ipfs.dag.get(pair.cid)).value;
+        const record = await this.ipfsService.retrieveRecord(pair.cid);
```

Comment:
What if either of these calls to `retrieveRecord` fails?

Reply:
Good point! There's a check for reachable CIDs and retries also. If this fails than the whole anchoring will fail too. If that starts to happen that would probably mean that there are some issues with the IPFS instance that we're running (cache, etc.).

Reply:
Yep that's fair, the earlier check for unreachable CIDs should filter out the most common issues, if this fails here it means the CID went from accessible to not in a short period of time, likely indicating major network flakiness, so failing the whole anchor if that happens is probably fine at least for now.  Longer term we'll want to be more resilient to transient network errors, but this is probably good enough for now.